metric/system/process/process.go

```diff
@@ -133,7 +134,7 @@ func (procStats *Stats) GetSelf() (ProcState, error) {
 	if err != nil {
 		return ProcState{}, fmt.Errorf("error fetching PID %d: %w", self, err)
 	}
-
+	procStats.ProcsMap[self] = pidStat
```

Comment:
What is this doing? I assume it is related to this in the PR description:

> Make processes fetched with GetSelf() persist across calls

Reply:
Yep. We need to save off the previous value in order to calculate CPU usage percentages.